automated install/basic-install.sh

```diff
@@ -776,77 +776,98 @@ It is also possible to use a DHCP reservation, but if you are going to do that,
     fi
 }
 
-# dhcpcd is very annoying,
+# configure networking via dhcpcd
 setDHCPCD() {
-    # but we can append these lines to dhcpcd.conf to enable a static IP
-    echo "interface ${PIHOLE_INTERFACE}
-    static ip_address=${IPV4_ADDRESS}
-    static routers=${IPv4gw}
-    static domain_name_servers=127.0.0.1" | tee -a /etc/dhcpcd.conf >/dev/null
+    # check if the IP is already in the file
+    if grep -q "${IPV4_ADDRESS}" /etc/dhcpcd.conf; then
+        printf "  %b Static IP already configured\\n" "${INFO}"
+    # If it's not,
+    else
+        # we can append these lines to dhcpcd.conf to enable a static IP
+        echo "interface ${PIHOLE_INTERFACE}
+        static ip_address=${IPV4_ADDRESS}
+        static routers=${IPv4gw}
+        static domain_name_servers=127.0.0.1" | tee -a /etc/dhcpcd.conf >/dev/null
+        # Then use the ip command to immediately set the new address
+        ip addr replace dev "${PIHOLE_INTERFACE}" "${IPV4_ADDRESS}"
+        # Also give a warning that the user may need to reboot their system
+        printf "  %b Set IP address to %s \\n  You may need to restart after the install is complete\\n" "${TICK}" "${IPV4_ADDRESS%/*}"
+    fi
 }
 
-setStaticIPv4() {
+# configure networking ifcfg-xxxx file found at /etc/sysconfig/network-scripts/
+# this function requires the full path of an ifcfg file passed as an argument
+setIFCFG() {
     # Local, named variables
     local IFCFG_FILE
     local IPADDR
     local CIDR
+    IFCFG_FILE=$1
+    printf -v IPADDR "%s" "${IPV4_ADDRESS%%/*}"
+    # check if the desired IP is already set
+    if grep -Eq "${IPADDR}(\\b|\\/)" "${IFCFG_FILE}"; then
+        printf "  %b Static IP already configured\\n" "${INFO}"
+    # Otherwise,
+    else
+        # Put the IP in variables without the CIDR notation
+        printf -v CIDR "%s" "${IPV4_ADDRESS##*/}"
+        # Backup existing interface configuration:
+        cp "${IFCFG_FILE}" "${IFCFG_FILE}".pihole.orig
+        # Build Interface configuration file using the GLOBAL variables we have
+        {
+        echo "# Configured via Pi-hole installer"
+        echo "DEVICE=$PIHOLE_INTERFACE"
+        echo "BOOTPROTO=none"
+        echo "ONBOOT=yes"
+        echo "IPADDR=$IPADDR"
+        echo "PREFIX=$CIDR"
+        echo "GATEWAY=$IPv4gw"
+        echo "DNS1=$PIHOLE_DNS_1"
+        echo "DNS2=$PIHOLE_DNS_2"
```

Comment:
The DHCPCD config uses 127.0.0.1 as the DNS server. Is there a reason that this is not done here?

Reply:
Do we want to force the device to use itself for DNS by default?  
I prefer the ability to stop pi-hole services and not affect the connectivity of my other hosted services. This also helps address when users have a failed install and can no longer update the OS or resolve any url's due to having the loopback set as the nameserver.
Typically pi-hole is setup as a headless service either alone or with other services.. perhaps we should look at adding another option for the user to decide if they would like their server to self-resolve?

Reply:
issue: https://github.com/pi-hole/pi-hole/issues/2535